File: NetFramework/Reports/C1Report/Cs/C1dView/C1dView.Settings.cs
/*
 * This file is part of C1Report's C1dView sample.
 * 
 * This file contains settings persistence code.
 * 
 * Copyright GrapeCity, Inc.
 */
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Drawing;
using C1.C1Preview;
using C1.Win.C1Preview;

namespace C1dView
{
    /// <summary>
    /// Encapsulates preview settings which can be persisted together
    /// as a single Settings.settings variable of type C1dView.C1dViewSettings.
    /// </summary>
    public class C1dViewSettings
    {
        #region Defaults and limits
        public static readonly Size MinMainWindowSize = new Size(200, 200);
        #endregion

        #region Settings
        public Rectangle MainWindowBounds { get; set; }
        public List<Pair<string, string>> RecentDocuments { get; set; }
        #endregion

        #region Methods
        /// <summary>
        /// Loads the last saved settings into the preview.
        /// </summary>
        /// <param name="previewForm">The preview form.</param>
        public static void Load(C1dViewForm previewForm)
        {
            try
            {
                var settings = Properties.Settings.Default;
                if (settings == null)
                    return;
                if (settings.C1dViewSettings == null)
                {
                    Save(previewForm);
                    return;
                }
                var vs = settings.C1dViewSettings;
                // main window position:
                if (vs.MainWindowBounds.Width >= MinMainWindowSize.Width &&
                    vs.MainWindowBounds.Height >= MinMainWindowSize.Height &&
                    vs.MainWindowBounds.X >= 0 && vs.MainWindowBounds.Y >= 0)
                {
                    previewForm.StartPosition = System.Windows.Forms.FormStartPosition.Manual;
                    previewForm.Bounds = vs.MainWindowBounds;
                }
                // recent documents:
                C1dViewForm.RecentDocuments.Clear();
                if (vs.RecentDocuments != null)
                    C1dViewForm.RecentDocuments.AddRange(vs.RecentDocuments);
            }
            catch
            {
                // ignore settings errors, not much to do here.
            }
        }

        /// <summary>
        /// Saves current settings.
        /// </summary>
        /// <param name="previewForm">The preview form.</param>
        public static void Save(C1dViewForm previewForm)
        {
            try
            {
                var settings = Properties.Settings.Default;
                if (settings == null)
                    return;
                if (settings.C1dViewSettings == null)
                    settings.C1dViewSettings = new C1dViewSettings();
                var vs = settings.C1dViewSettings;
                // main window position:
                vs.MainWindowBounds = previewForm.Bounds;
                // recent documents:
                vs.RecentDocuments = new List<Pair<string, string>>();
                vs.RecentDocuments.AddRange(C1dViewForm.RecentDocuments);
                //
                settings.Save();
            }
            catch
            {
                // ignore settings errors, not much to do here.
            }
        }
        #endregion
    }
}

File: ExcludedSamples/Reports/C1Report/VB/C1dView/C1dViewSettings.vb
''
'' This file is part of C1Report's C1dView sample.
''
'' This file contains settings persistence code.
''
''Copyright GrapeCity, Inc.
''
Imports System
Imports System.Collections.Generic
Imports System.Linq
Imports System.Text
Imports System.Drawing
Imports C1.C1Preview
Imports C1.Win.C1Preview

''' <summary>
''' Encapsulates preview settings which can be persisted together
''' as a single Settings.settings variable of type C1dView.C1dViewSettings.
''' </summary>
Public Class C1dViewSettings
#Region "Defaults and limits"
    Public Shared ReadOnly MinMainWindowSize As Size = New Size(200, 200)
#End Region

#Region "Settings"
    Public Property MainWindowBounds As Rectangle
    Public Property RecentDocuments As List(Of Pair(Of String, String))
#End Region

#If True Then
#Region "Methods"
    ''' <summary>
    ''' Loads the last saved settings into the preview.
    ''' </summary>
    ''' <param name="previewForm">The preview form.</param>
    Public Shared Sub Load(ByVal previewForm As C1dViewForm)

        Try
            Dim settings = My.MySettings.Default
            If (settings Is Nothing) Then
                Return
            End If
            If (settings.C1dViewSettings Is Nothing) Then
                Save(previewForm)
                Return
            End If
            Dim vs = settings.C1dViewSettings
            '' main window position:
            If (vs.MainWindowBounds.Width >= MinMainWindowSize.Width AndAlso _
                vs.MainWindowBounds.Height >= MinMainWindowSize.Height AndAlso _
                vs.MainWindowBounds.X >= 0 AndAlso _
                vs.MainWindowBounds.Y >= 0) Then
                previewForm.StartPosition = System.Windows.Forms.FormStartPosition.Manual
                previewForm.Bounds = vs.MainWindowBounds
            End If
            '' recent documents:
            C1dViewForm.RecentDocuments.Clear()
            If (vs.RecentDocuments IsNot Nothing) Then
                C1dViewForm.RecentDocuments.AddRange(vs.RecentDocuments)
            End If
        Catch
            '' ignore settings errors, not much to do here.
        End Try
    End Sub

    ''' <summary>
    ''' Saves current settings.
    ''' </summary>
    ''' <param name="previewForm">The preview form.</param>
    Public Shared Sub Save(ByVal previewForm As C1dViewForm)

        Try

            Dim settings = My.MySettings.Default
            If (settings Is Nothing) Then
                Return
            End If
            If (settings.C1dViewSettings Is Nothing) Then
                settings.C1dViewSettings = New C1dViewSettings()
            End If
            Dim vs = settings.C1dViewSettings
            '' main window position:
            vs.MainWindowBounds = previewForm.Bounds
            '' recent documents:
            vs.RecentDocuments = New List(Of Pair(Of String, String))()
            vs.RecentDocuments.AddRange(C1dViewForm.RecentDocuments)
            ''
            settings.Save()

        Catch
            '' ignore settings errors, not much to do here.
        End Try
    End Sub
#End Region
#End If
End Class
